### PR TITLE
make tickets required for PRs

### PR DESCRIPTION
**Pull request by mogoodrich**
Start the discussion here... :)

### CODE REVIEW

**Comment by mogoodrich** [comment]
We also could change this to explain the limited cases where no ticket is necessary... IMO, beyond just features, any change that requires post-commit testing should be tickted.

**Comment by denniskigen** [comment]
Thanks for setting this in motion, @mogoodrich. This is the typical PR template we use across our repos:

CleanShot 2024-03-25 at 11  58 47@2x

Do we want to mandate that:

- The ticket number is included in the PR title and a conventional commit label i.e., something like: `(fix) O3-2977: Make end date field editable for inactive conditions`.
- A link to the ticket is included in the `Related issue` section of the PR template.

**Comment by denniskigen** [comment]
> We also could change this to explain the limited cases where no ticket is necessary... IMO, beyond just features, any change that requires post-commit testing should be ticked.

I agree.

**Comment by ibacher** [comment]
@denniskigen Yes to both of your points.

@mogoodrich Do we have a good sense of what requires "post-commit" testing? As I've seen it, that's primarily a category for things we've done that are tested, but only locally by the dev, i.e., most things. I'm asking because if we're going to have guidance it needs to be clear. Otherwise, I'd just stick to "Every commit must have a ticket".

**Comment by mogoodrich** [comment]
Yes, almost everything requires post-commit testing @ibacher ... not sure if we canclearly can define what the exceptions are.  Some examples: auto-commtting translations from Transifex,fixing typos in README or comments.  I'd be fine with sticking with "every commit must have a ticket" in the docs and just trusting that senior devs know the (very) few times they can break that convention.

**Comment by brandones** [comment]
Just echoing what I said in the PR (which @mseaton +1'd):

I would suggest we don't require tickets for (chore) PRs.

See for reference the [core release notes](https://github.com/openmrs/openmrs-esm-core/releases/tag/v5.5.0).

Seems like we're gravitating toward requiring tickets for (fix) PRs for the benefit of post-commit testing. I'm not clear that we actually have a meaningful post-commit testing process, but if we do, and if tickets are part of that, that's great.

File: pages/docs/frontend-modules/contributing.en-US.mdx [hunk 1]
```diff
@@ -7,7 +7,7 @@ If you’re interested in contributing to O3, you’ll need to take the followin
 
 ### Contributing guidelines
 
-- **Pull requests (PRs) are not required to correspond to a ticket** - If there is a ticket for the PR, it should be clearly named and linked in the PR.
+- **Pull requests (PRs) are required to correspond to a ticket** -  The ticket related to the PR should be clearly named and linked in the PR.
```
Comment (mogoodrich) [inline pages/docs/frontend-modules/contributing.en-US.mdx:10]:
```suggestion
- **Pull requests (PRs) of type "feature", "fix", "refactor", "perf", and "BREAKING" (see "type of change" section below) are required to correspond to a ticket** -  The ticket related to the PR should be clearly named and linked in the PR.
```

**Comment by mogoodrich** [comment]
Makes sense @brandones  that chores don't need a ticket... and, in fact, good point, the ticket types are good way to make a distinction, see my suggest on my own change above.

And, yeah, we *should* be moving toward more meaningful post-commit testing.

I included refactor and perf in the list of the types that should require a commit, because basically anything that could potentially break something should have a ticket.   (Basically, my reasoning here is that if I discover a bug in the code and trace it down to a specific commit, I want there to already be an existing ticket that I can comment on about the problem, it shouldn't be the burden of the discoverer of a bug based on a refactor to create the ticket).

**Review by ibacher — APPROVED** [review]

**Review by denniskigen — APPROVED** [review]
LGTM. Thanks, @mogoodrich.

**Pull request merged**